File: util/randvect.py
#!/usr/bin/env python3

import sys
import random

size = None
minv = None
maxv = None

try:
    size = int(sys.argv[1])
    minv = int(sys.argv[2])
    maxv = int(sys.argv[3])
except IndexError:
    sys.stderr.write('Required: size, min max\n')
except ValueError as e:
    sys.stderr.write('Invalid argument: {}\n'.format(e))

if size is None or minv is None or maxv is None:
    sys.exit(1)

vector = [random.uniform(minv,maxv) for __ in range(size)]
random.shuffle(vector)

sys.stdout.write('\n'.join(str(x) for x in vector))
sys.stdout.flush()

File: util/randvect.c
#include <stdio.h>
#include <stdlib.h>

double get_rand_real(int min, int max) {
    return min + ( rand() / (double) RAND_MAX ) * (max - min);
}

int main(int argc, char *argv[]) {

    long int i, size;
    int min, max;
    double randreal;

    if (argc != 4) {
        printf("required args: size min max\n");
        exit(1);
    }

    size = atol(argv[1]);
    min = atoi(argv[2]);
    max = atoi(argv[3]);

    for (i=0; i<size; i++)
        printf("%.20f\n", get_rand_real(min,max));

    return 0;
}
